### ISSUE TITLE
Checksum error while performing update and then rollback

### ISSUE THREAD

**Issue by ebussieres** [liquibase/liquibase#5278]
### Search first

- [X] I searched and no similar issues were found

### Description

I am using spring boot and a liquibase parameter in a sql script. 

When setting the spring flag `test-rollback-on-update` to true, liquibase performs an update and then a rollback. There's a checksum validation error when performing the rollback step. It seems to be related when you use a liquibase parameter in a sql script 

### Steps To Reproduce

There's a [demo ](https://github.com/ebussieres/spring-liquibase-bug) available here which demonstrate the regression. Just run the [test ](https://github.com/ebussieres/spring-liquibase-bug/blob/master/src/test/java/com/example/demo/DemoApplicationTests.java) to see the problem.

### Expected/Desired Behavior

No checksum validation error

### Liquibase Version

4.24

### Database Vendor & Version

H2

### Liquibase Integration

Spring boot 3.2.0

### OS and/or Infrastructure Type/Provider

Windows 10

### Additional Context

An issue has been submitted to [spring](https://github.com/spring-projects/spring-boot/issues/38522). They asked me to submit an issue here.

### Are you willing to submit a PR?

- [ X ] I'm willing to submit a PR (Thank you!)

**Comment by wilkinsona** [comment]
Here's a more minimal reproducer that largely takes Spring out of the picture:

```java
JdbcDataSource dataSource = new JdbcDataSource();
dataSource.setUrl("jdbc:h2:mem:test");
SpringLiquibase springLiquibase = new SpringLiquibase();
springLiquibase.setChangeLog("classpath:db/changelog/db.changelog-master.yaml");
springLiquibase.setDataSource(dataSource);
springLiquibase.setTestRollbackOnUpdate(true);
springLiquibase.setChangeLogParameters(Map.of("foo", "bar"));
springLiquibase.afterPropertiesSet();
```

**Comment by EvaristeGalois11** [comment]
After a git bisect the bug seems to come from this PR https://github.com/liquibase/liquibase/pull/4653

I will be able to look more into it during the weekend

**Comment by ebussieres** [comment]
It seems to fix the problem if you add the CHANGELOG_ARG to rollback command in Liquibase class

```
runInScope(() ->
            new CommandScope(RollbackCountCommandStep.COMMAND_NAME)
            ...
           .addArgumentValue(DatabaseChangelogCommandStep.CHANGELOG_ARG, databaseChangeLog)
            ...
            .execute()
);
```

**Comment by ebussieres** [comment]
@EvaristeGalois11 : Just made a PR to solve the problem. Feel free to use it as an initial draft.

**Comment by EvaristeGalois11** [comment]
> It seems to fix the problem if you add the CHANGELOG_ARG to rollback command in Liquibase class
> 
> ```
> runInScope(() ->
>             new CommandScope(RollbackCountCommandStep.COMMAND_NAME)
>             ...
>            .addArgumentValue(DatabaseChangelogCommandStep.CHANGELOG_ARG, databaseChangeLog)
>             ...
>             .execute()
> );
> ```

Well that was quick :D

**Comment by ebussieres** [comment]
Fixed via PR #5283

**Comment by bbortt** [comment]
this hasn't been release yet, right?

**Comment by ebussieres** [comment]
@bbortt It has been fixed in version 4.25.1

**Comment by bbortt** [comment]
ok, sry. my bad. version 4.20.0 was still forced from one of my boms.
